gcapi/create_strategies.py

```
@@ -247,10 +334,19 @@ def select_socket_value_strategy(  # noqa: C901
             )
     elif socket.super_kind.casefold() == "image":
         if not isinstance(spec.files, UnsetType):
-            return ImageCreateStrategy(
-                files=spec.files,
-                **kwargs,
-            )
+            if socket.kind.casefold() == "dicom image set" and not isinstance(
```

If it's a dicom kind socket, it always needs to become a dicom image set. A regular image should not get accepted for a dicom kind socket anyway, so I think you can stop and throw an error if image name is not set?

---

Ah, of course!